### PR TITLE
188 single col fix@main

### LINKED ISSUE #188: pagination resulting in single column fails.
related to [rlistings issue ](https://github.com/insightsengineering/rlistings/issues/143)

returns an error where spans and strings dimensions are transposed.

### PR DESCRIPTION
**Pull request by ayogasekaram**
closes #188 
related to [rlistings issue](https://github.com/insightsengineering/rlistings/issues/143)

### CODE REVIEW

**Review by shajoezhu — COMMENTED** [review]
hi @ayogasekaram 
can you update the news please

File: R/matrix_form.R [hunk 1]
```diff
@@ -880,8 +880,12 @@ truncate_one_span <- function(spanrow, j) {
 }
 
 truncate_spans <- function(spans, j) {
+  if (length(spans[1,])==1){
+    as.matrix(apply(spans, 1, truncate_one_span, j = j))
+  }
+  else
     t(apply(spans, 1, truncate_one_span, j = j))
-}
+  }
```
Comment (shajoezhu) [inline R/matrix_form.R:888]: remove space

File: NEWS.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ## formatters 0.5.1.9003
+ * ``truncate_spans` function modified to handle matrix with a single column after pagination.
```
Comment (Melkiades) [inline NEWS.md:2]:
```suggestion
 * `truncate_spans` function modified to handle matrix with a single column after pagination.
```

File: R/matrix_form.R [hunk 3]
```diff
@@ -880,6 +880,10 @@ truncate_one_span <- function(spanrow, j) {
 }
 
 truncate_spans <- function(spans, j) {
+  if (length(spans[1,])==1){
```
Comment (Melkiades) [inline R/matrix_form.R:883]:
```suggestion
  if (length(spans[1,]) == 1){
```

File: tests/testthat/test-pagination.R [hunk 4]
```diff
@@ -316,3 +316,16 @@ test_that("non-monospaced fonts are caught", {
         )
     )
 })
+
+test_that("spans and string matrix match after pagination
+          when table has single column", {
+
+  df <- as.data.frame(mtcars[,c(1)])
```
Comment (Melkiades) [inline tests/testthat/test-pagination.R:323]:
```suggestion
  df <- as.data.frame(mtcars[, 1])
```

File: tests/testthat/test-pagination.R [hunk 5]
```diff
@@ -316,3 +316,16 @@ test_that("non-monospaced fonts are caught", {
...
+  test <- formatters:::basic_matrix_form(df)
+  pag_test <- formatters:::paginate_to_mpfs(test)
+
+  expect_identical(
+    dim(pag_test[[1]]$spans), dim(pag_test[[1]]$strings))
+})
+
```
Comment (Melkiades) [inline tests/testthat/test-pagination.R:330]:
```suggestion
```

File: tests/testthat/test-pagination.R [hunk 6]
```diff
@@ -316,3 +316,15 @@ test_that("non-monospaced fonts are caught", {
         )
     )
 })
+
+test_that("spans and string matrix match after pagination
+          when table has single column", {
+
+  df <- as.data.frame(mtcars[, 1])
+  test <- formatters:::basic_matrix_form(df)
+  pag_test <- formatters:::paginate_to_mpfs(test)
```
Comment (Melkiades) [inline tests/testthat/test-pagination.R:325]:
```suggestion
  test <- basic_matrix_form(df)
  pag_test <- paginate_to_mpfs(test)
```
you do not need `:::` for internals in testthat

**Review by Melkiades — APPROVED** [review]
LGTM! Thanks @ayogasekaram for solving this ;)

**Comment by ayogasekaram** [comment]
rtables, rlistings and tern tests pass with this change

File: NEWS.md [hunk 7]
```diff
@@ -1,4 +1,5 @@
 ## formatters 0.5.1.9003
+ * `truncate_spans` function modified to handle matrix with a single column after pagination.
```
Comment (gmbecker) [inline NEWS.md:2]: truncate spans is a internal function, please modify the news entry to talk about the user-facing bug that was fixed instead.

**Pull request merged**